src/osd/PrimaryLogPG.cc

```diff
@@ -5518,7 +5518,12 @@ int PrimaryLogPG::do_read(OpContext *ctx, OSDOp& osd_op) {
     }
     if (r >= 0)
       op.extent.length = r;
-    else {
+    else if (r == -EAGAIN) {
```

Comment:
under what circumstances, would the sync read return `-EAGAIN`?

Reply:
@thinkercui I have the same question as Kefu. Seems EAGAIN is not a valid return code for sync-read.  Would you mind elaborating a bit?

Reply:
> @thinkercui I have the same question as Kefu. Seems EAGAIN is not a valid return code for sync-read. Would you mind elaborating a bit?

@tchaikov  @xiexingguo  thanks. sync-read may return `-EIO` and this will make `PrimaryLogPG::rep_repair_primary_object` return '-EAGAIN' which may change the return code to `-EAGAIN`.

Reply:
Ahh, that's right